src/fdc_ts/core.clj

```diff
@@ -51,19 +51,23 @@
 
 ;TODO Move put to separate module
 
-(defn- auth [token ctx]
-  (= (token env) (get-in ctx [:request :headers "auth-token"])))
+(defn- auth [token project-tokens project ctx]
+  (def request-token  (get-in ctx [:request :headers "auth-token"]))
+  (or (= (token env) request-token)
+      (= (get-in (json/parse-string (project-tokens env))  [project]) request-token)))
 
 (defn- auth-configured [token ctx]
   (token env))
 
-(def auth-publish (partial auth :auth-token-publish))
-(def auth-publish-configured (partial auth-configured :auth-token-publish))
+(def auth-publish (partial auth :auth-token-publish :auth-token-project))
+(def auth-publish-configured (and(partial auth-configured :auth-token-publish)
+                                 (partial auth-configured :auth-token-project)))
 
-(def auth-statistics (partial auth :auth-token-statistics))
-(def auth-statistics-configured (partial auth-configured :auth-token-statistics))
+(def auth-statistics (partial auth :auth-token-statistics :auth-token-project))
+(def auth-statistics-configured (and (partial auth-configured :auth-token-statistics)
```

Comment:
do not understand the (and ...) here

Reply:
Here, we check if the project tokens field is defined in the env. 
auth-statistics-configured is used to see if the service is available.